src/ape/api/address.py

```diff
@@ -171,15 +170,33 @@ def codesize(self) -> int:
         The number of bytes in the smart contract.
         """
 
-        return len(self.code)
+        code = self.code
+        return len(code) if isinstance(code, bytes) else len(bytes.fromhex(code.lstrip("0x")))
 
     @property
     def is_contract(self) -> bool:
         """
         ``True`` when there is code associated with the address.
         """
 
-        return len(HexBytes(self.code)) > 0
+        return self.codesize > 0
+
+    @property
+    def delegate(self) -> Optional["ContractInstance"]:
+        """
+        Check and see if Account has a "delegate" contract, which is a contract that this account
+        delegates functionality to. This could be from many contexts, such as a Smart Wallet like
+        Safe (https://github.com/ApeWorX/ape-safe) which has a Singleton class it forwards to, or
+        an EOA using a EIP7702-style delegate. Returning `None` means that the account does not
+        have a delegate avaiable, which should be overrided in plugins that support ecosystems
+        where such methods are possible natively (e.g. EIP7702 context), or plugins such as smart
+        wallets which have this functionality. The default implementation is to return `None`.
```

Comment:
```suggestion
        wallets which have this functionality. The default implementation is to return ``None``.
```

Reply:
Actually have to rewrite this entire doc now that I am delegating to `ChainManager.get_delegate`